projects/core/src/lib/operators/dialogs/raster-kernel/raster-kernel.component.html

```diff
@@ -0,0 +1,116 @@
+<geoengine-sidenav-header>Raster Kernel</geoengine-sidenav-header>
+
+<geoengine-dialog-help>
+    <p>
+        The <i>Raster Kernel</i> operator computes a rectangular kernel for a pixel and its neighborhood. The operator can be defined as a
+        matrix kernel with weights or as specialized kernel types, where only the size is specified.
+    </p>
+    <p>You can find more information at <a href="https://docs.geoengine.io/operators/rasterkernel.html">docs.geoengine.io</a>.</p>
+</geoengine-dialog-help>
+
+<form [formGroup]="form" (ngSubmit)="add()" fxLayout="column" *ngIf="projectHasRasterLayers$ | async; else noDataInfo">
+    <div fxFlex class="container">
+        <geoengine-layer-selection [types]="RASTER_TYPE" formControlName="rasterLayer"></geoengine-layer-selection>
+        <geoengine-dialog-section-heading title="Configuration" subtitle="Specify the operator"></geoengine-dialog-section-heading>
+
+        <mat-form-field appearance="fill">
+            <mat-label>Kernel Type</mat-label>
+            <mat-select formControlName="kernelType">
+                <mat-option value="convolution">Convolution</mat-option>
+                <mat-option value="standardDeviation">Standard Deviation</mat-option>
+            </mat-select>
+        </mat-form-field>
+
+        <ng-container [ngSwitch]="form.controls.kernelType.value">
+            <ng-container *ngSwitchCase="'convolution'">
+                <div class="matrix-controls">
+                    <button mat-button color="primary" type="button" (click)="enlargeMatrix()">
```

Comment:
what about non-quadratic matrices? Are they not important enough for now?

Reply:
I would leave it out for now.